### PR TITLE
Misc TINC cleanup

### PR DESCRIPTION
**Pull request by hlinnaka**
Now that the TINC suite is in the git repository, the fun of cleaning it all up so that there are no useless or redundant tests, begins.

@jimmyyih, are you continuing to work on this? Can you actually run the tests?

### CODE REVIEW

File: src/test/tinc/tincrepo/mpp/gpdb/tests/catalog/schema_topology/test_ST_DMLOverJoinsTest.py [hunk 1]
```diff
@@ -1,103 +0,0 @@
-"""
```
Comment (asimrp) [inline src/test/tinc/tincrepo/mpp/gpdb/tests/catalog/schema_topology/test_ST_DMLOverJoinsTest.py]: Removal of this file will break `tests/storage/crashrecovery/test_crash_recovery_schema_topology.py`.

**Comment by asimrp** [comment]
Removing tests in catalog/schema_topology seems to impact on the following tests:
```
tincrepo/mpp/gpdb/tests]$ find . -name '*.py' -exec grep -l -e 'catalog.schema_topology' {} \;
./catalog/schema_topology/test_ST_EnhancedTableFunctionTest.py
./storage/crashrecovery/test_crash_recovery_schema_topology.py
./utilities/gpexpand/test_gpexpand.py
./utilities/recoverseg/gprecoverseg_tests/test_gprecoverseg.py
```

**Comment by Eulerizeit** [comment]
Is there a need/desire to parse these tests into smaller suites with fewer
dependencies?

**Comment by hlinnaka** [comment]
On 08/11/2016 08:27 PM, Robert Eckhardt wrote:
> Is there a need/desire to parse these tests into smaller suites with fewer
> dependencies?

Yes.

- Heikki

File: src/test/tinc/tincrepo/mpp/gpdb/tests/catalog/schema_topology/test_ST_DMLOverJoinsTest.py (hunk 1 above)
Comment (hlinnaka) [inline src/test/tinc/tincrepo/mpp/gpdb/tests/catalog/schema_topology/test_ST_DMLOverJoinsTest.py]: On 08/11/2016 08:20 PM, Asim RP wrote:
>> @@ -1,103 +0,0 @@
>> -"""
>
> Removal of this file will break `tests/storage/crashrecovery/test_crash_recovery_schema_topology.py`.

Hmm. The DML_Over_Joins test doesn't seem very interesting from a crash 
recovery point of view. An UPDATE that uses a join isn't any different 
from other kinds of updates, as far as storage and crash recovery are 
concerned. How about we just remove the DMLOverJoinsTest from 
tests/storage/crashrecovery/test_crash_recovery_schema_topology.py as well?

The other dependent tests, especially 
mpp.gpdb.tests.catalog.schema_topology.test_ST_GPFilespaceTablespaceTest.GPFilespaceTablespaceTest, 
seem more interesting.

- Heikki

**Comment by jimmyyih** [comment]
Everything looks fine except for the concern on 267f7bf15abc40303708ceb0645d0327b05ddb19 removing DML_over_joins test as noted by Asim.

The dependencies:
```
storage/crashrecovery/test_crash_recovery_schema_topology.py:47:        classlist.append('mpp.gpdb.tests.catalog.schema_topology.test_ST_DMLOverJoinsTest.DMLOverJoinsTest')
utilities/gpexpand/test_gpexpand.py:499:        classlist.append('mpp.gpdb.tests.catalog.schema_topology.test_ST_DMLOverJoinsTest.DMLOverJoinsTest')
utilities/recoverseg/gprecoverseg_tests/test_gprecoverseg.py:159:        test_case_list4.append('mpp.gpdb.tests.catalog.schema_topology.test_ST_DMLOverJoinsTest.DMLOverJoinsTest')
```

@Chibin, can you check if gpexpand tests actually need it?

**Comment by Chibin** [comment]
At first glance, it doesn't seem like something that gpexpand necessarily needs.

**Comment by jimmyyih** [comment]
+1 after removing the three dependent lines.

**Comment by hlinnaka** [comment]
On 08/15/2016 09:35 PM, Jimmy Yih wrote:
> +1 after removing the three dependent lines.

@jimmyyih, since you know how to actually run these tests, and you're 
hacking in this area anyway, could you make the required changes and 
push this, please?

- Heikki

**Comment by jimmyyih** [comment]
@hlinnaka, yeah I can help make the changes and push this in.  Do you want your commits squashed in some way or at all?

**Comment by hlinnaka** [comment]
On 08/18/2016 12:01 AM, Jimmy Yih wrote:
> @hlinnaka, yeah I can help make the changes and push this in.  Do you
> want your commits squashed in some way or at all?

Thanks! I think the commits are fine as they are, but I'll let you make 
the final call.

- Heikki

**Comment by jimmyyih** [comment]
Pushed into master as 8 commits:
https://github.com/greenplum-db/gpdb/commit/fcbfd412c9ec1c1c5000174728b810e2d1baee1a
https://github.com/greenplum-db/gpdb/commit/6a33d211860f83eb49a228730f42530e7dcda395
https://github.com/greenplum-db/gpdb/commit/0a084e0e3c9269c4c1c8867ee374055113c3130f
https://github.com/greenplum-db/gpdb/commit/583c4be1de0719e33af0b7ed713b00c9b6911e91
https://github.com/greenplum-db/gpdb/commit/43a0859802b91d6552ed619fa23c63181d19a79d
https://github.com/greenplum-db/gpdb/commit/a873b4639a60e4dee5d8c82bf00331b2ad68d2d4
https://github.com/greenplum-db/gpdb/commit/0937cd2f67d0ce480650244b5592eee1e2bb24c9
https://github.com/greenplum-db/gpdb/commit/1eeea564ca29f72562d427e7bba355d6f7f229f1

**Pull request closed without merging**